feugy/tabulous
# feat(web): collapsible hand

### What's in there?

Meshes (like cards) have different dimensions, depending on their game. Boards, in particular, may have very different sizes.
We need to adjust zoom ranges and positions (both main and hand scene), based on the loaded game.

The hand overlay should also be collapsible. By reusing `MinimizableSection` component, we can also make it resizable

Are included here:

- feat(web): introduces collapsible and resizable game hand overlay
- feat(web): applies per-game custom zoom level and ranges
- style(web): moves camera switch to the top, to avoid conflicting with hand overlay controls
- refactor(web): hand manager now uses external overlay element
- feat(server): includes zoom ranges per game descriptor
- feat(coinche): uses custom hand zoom level

### How to test?

Any games supporting player hands, but in particular, Coinche.
A few things to try out:
- browse between Coinche and Klondike games to check that the hand only appears on Coinche.
- on a new Coinche game, the hand starts minimized. Drawing a mesh should maximize it. Playing the last drawn mesh should minimize it again
- on a game with hand meshes, the hand overlay can be minimized, restored, and resized.

![collapsible-hand small](https://user-images.githubusercontent.com/186268/165992241-9a2d1065-dc49-4bb5-bf1e-f8782a196a34.gif)
